File: C++/stack/stack.cpp
#include <iostream>
#include <cstdlib> // for exit()
using namespace std;

int stack[50];
int top = -1;

void push()
{
    if (top == 49)
        cout << "\nOverflow\n";
    else
    {
        top++;
        cout << "Enter the new element: ";
        cin >> stack[top];
    }
}

void pop()
{
    if (top == -1)
        cout << "\nUnderflow\n";
    else
    {
        cout << "Popped element: " << stack[top] << endl;
        top--;
    }
}

void peek()
{
    if (top == -1)
        cout << "\nUnderflow\n";
    else
        cout << stack[top] << " is on top\n";
}

void display()
{
    if (top == -1)
        cout << "\nUnderflow\n";
    else
    {
        cout << "Stack elements (top to bottom): ";
        for (int i = top; i >= 0; i--)
        {
            cout << stack[i] << " ";
        }
        cout << endl;
    }
}

int main()
{
    int ch;
    while (true)
    {
        cout << "\n1: Push\n2: Pop\n3: Peek\n4: Display\n5: Exit\nChoose one option: ";
        cin >> ch;
        switch (ch)
        {
        case 1:
            push();
            break;
        case 2:
            pop();
            break;
        case 3:
            peek();
            break;
        case 4:
            display();
            break;
        case 5:
            exit(0);
        default:
            cout << "Invalid input\n";
        }
    }
    return 0;
}

File: Python/stack/stack.py
stack = []
MAX_SIZE = 50

def push():
    if len(stack) >= MAX_SIZE:
        print("Overflow")
    else:
        value = int(input("Enter the new element: "))
        stack.append(value)

def pop():
    if not stack:
        print("Underflow")
    else:
        print("Popped element:", stack.pop())

def peek():
    if not stack:
        print("Underflow")
    else:
        print(stack[-1], "is on top")

def display():
    if not stack:
        print("Underflow")
    else:
        print("Stack elements (top to bottom):", *reversed(stack))

# Menu-driven logic without main()
while True:
    print("\n1: Push\n2: Pop\n3: Peek\n4: Display\n5: Exit")
    ch = int(input("Choose one option: "))
    
    if ch == 1:
        push()
    elif ch == 2:
        pop()
    elif ch == 3:
        peek()
    elif ch == 4:
        display()
    elif ch == 5:
        exit()
    else:
        print("Invalid input, please enter a valid option.")
